### PR TITLE
Add tests for member pruning [HZ-1917]

### PR DESCRIPTION
**Pull request by sumnerib**
Provides additional test coverage for member pruning functionality. The existing tests only use internal APIs to execute "prunable" SQL queries, whereas the added `MemberPruningTest` uses the public API.

Also, includes an additional case for
`AttributePartitioningStrategyIntegrationTest` for IDS type keys.

### CODE REVIEW

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/MemberPruningTest.java [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    private void configureMapWithAttributes(String mapName, String... attributes) {
+        MapConfig mc = new MapConfig(mapName);
+        for (String attribute : attributes) {
+            mc.getPartitioningAttributeConfigs()
+                    .add(new PartitioningAttributeConfig(attribute));
+        }
+        instance().getConfig().addMapConfig(mc);
+    }
+
+    private void assertInvokedOnlyOnMembers(HazelcastInstance... members) {
```
Comment (k-jamroz) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/MemberPruningTest.java:193]: such methods and listener are in `SqlEndToEndTestSupport`. you could use it as base class and move `assertInvokedOnlyOnMembers` there

File: hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/MemberPruningTest.java [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toSet;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class MemberPruningTest extends SqlTestSupport {
+
+    public static final int MEMBER_COUNT = 5;
+    private static final JobInvocationObserverImpl JOB_INVOCATION_OBSERVER = new JobInvocationObserverImpl();
```
Comment (k-jamroz) [inline hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/MemberPruningTest.java:56]: `SqlEndToEndTestSupport` registers new `jobInvocationObserver` instance for each test method, but does not unregister previous ones. This is not harmful, but could be improved. Here you share the observer between tests, which might make them dependent

**Comment by sumnerib** [comment]
run-lab-run

**Comment by k-jamroz** [comment]
I will fix `DistributedObjectDestroyedException` in separate PR

**Review by k-jamroz — APPROVED** [review]

**Review by Fly-Style — APPROVED** [review]

**Comment by sumnerib** [comment]
run-lab-run

**Pull request merged**